piksi_tools/console/settings_view.py

```diff
@@ -223,34 +199,39 @@ def revert_to_prior_value(self, name, old, new, error_value=SBP_WRITE_STATUS.TIM
                  "   Error Value: {2}")
         invalid_setting_prompt.text = invalid_setting_prompt.text.format(self.name, new, error_value)
         invalid_setting_prompt.run()
+        self.reverting = False
+
+    def _write_value(self, old, new):
+        if (old != new and old is not Undefined and new is not Undefined):
+            self.confirmed_set = False
+            res = self.settings.settings_api.write(self.section, self.name, new)
```

Comment:
Does this just block while the api waits for a response?  What does that do to the UI?  Have you tested over a lossy connection, or a slow connection?

Reply:
I have done some quick testing now on MacOs and see that it seems responsive when changing a setting even over a lossy connection, so I am satisfied.

Reply:
All reads and writes happen in separate `WorkQueue` threads leaving the GUI free to do its job.